blockchain/state_manager/src/lib.rs

```diff
@@ -41,23 +42,45 @@ where
             .ok_or_else(|| Error::ActorStateNotFound(actor.state.to_string()))?;
         Ok(act)
     }
-    /// Returns the epoch at which the miner was slashed at
-    pub fn miner_slashed(&self, _addr: &Address, _ts: &Tipset) -> Result<u64, Error> {
-        // TODO update to use power actor if needed
-        todo!()
+    fn get_actor(&self, addr: &Address, state_cid: &Cid) -> Result<Option<ActorState>, Error> {
+        let state =
+            HamtStateTree::new_from_root(self.bs.as_ref(), state_cid).map_err(Error::State)?;
+        state.get_actor(addr).map_err(Error::State)
     }
-    /// Returns the amount of space in each sector committed to the network by this miner
-    pub fn miner_sector_size(&self, addr: &Address, ts: &Tipset) -> Result<SectorSize, Error> {
-        let act: miner::State = self.load_actor_state(addr, ts)?;
-        // * Switch back to retrieving from Cid if/when changed in actors
-        // let info: miner::MinerInfo = self.bs.get(&act.info)?.ok_or_else(|| {
-        //     Error::State("Could not retrieve miner info from IPLD store".to_owned())
-        // })?;
-        Ok(act.info.sector_size)
+    /// Returns true if miner has been slashed or is considered invalid
+    pub fn is_miner_slashed(&self, addr: &Address, state_cid: &Cid) -> Result<bool, Error> {
+        let ms: miner::State = self.load_actor_state(addr, state_cid)?;
+        if ms.post_state.has_failed_post() {
+            return Ok(true);
+        }
+
+        let ps: power::State = self.load_actor_state(&*STORAGE_POWER_ACTOR_ADDR, state_cid)?;
+        match ps.get_claim(self.bs.as_ref(), addr)? {
+            Some(_) => Ok(false),
+            None => Ok(true),
+        }
     }
-    pub fn get_actor(&self, addr: &Address, ts: &Tipset) -> Result<Option<ActorState>, Error> {
-        let state = HamtStateTree::new_from_root(self.bs.as_ref(), ts.parent_state())
-            .map_err(Error::State)?;
-        state.get_actor(addr).map_err(Error::State)
+    /// Returns raw address of a miner
+    pub fn get_miner_wrk_addr(&self, state_cid: &Cid, addr: &Address) -> Result<Address, Error> {
+        let ms: miner::State = self.load_actor_state(addr, state_cid)?;
+
+        let state =
+            HamtStateTree::new_from_root(self.bs.as_ref(), state_cid).map_err(Error::State)?;
+        // Note: miner::State info likely to be changed to CID
+        let addr = resolve_to_key_addr(&state, self.bs.as_ref(), &ms.info.worker)
+            .map_err(|e| Error::Other(format!("Failed to resolve key address; error: {}", e)))?;
+        Ok(addr)
+    }
+    /// Returns specified actor's claimed power and total network power as a tuple
+    pub fn get_power(&self, state_cid: &Cid, addr: &Address) -> Result<(BigUint, BigUint), Error> {
+        let ps: power::State = self.load_actor_state(addr, state_cid)?;
```

Comment:
Almost certain you want to use the builtin power actor address here, pretty sure this will never be pulled from an address being passed in? Might want to double check this (miner address state will not be in the form of a power actor state)